requirements/specifications/openrpc-extensions/app-passthrough-apis.md

```diff
@@ -0,0 +1,533 @@
+Firebolt events have a synchronous subscriber registration method, e.g. `Lifecycle.onInactive(true)`, in addition to asynchronous notifications when the event actually happens. For events powered by an app pass-through, only the asynchronous notifications are passed in by the providing app. The initial event registration is handled by the platform, and the success response is not handled by the providing app.
+
+This section only applies to platform methods that have an `event` tag.
+
+App provided event registration **MUST** not return an availability error due to a lack of providers, since one may be launched at a future point.
+
+**TODO**: do we want to search the catalog and return unavailable if none are possible? Seems overkill.
+
+To ensure that event provider methods all behave the same the provider method **MUST** have a `result` schema with `"type"` set to `"null"`, since it will not expect any data in the response from the platform after pushing the notification.
+
+The platform method result schema **MUST** either:
+
+> Match the *last* parameter schema on the provider method so that the result can be passed through.
+>
+> Have a property that matches the *last* parameter schema on the provider method so that the result can be passed through.
+
+**TODO**: Find event context params spec and move into this branch
+
+Example platform method with context:
+```json
+{
+    "name": "onFoo",
+    "tags": [
+        {
+            "name": "capabilities",
+            "x-uses": [
+                "xrn:firebolt:capabilities:example:foo"
+            ],
+            "x-provided-by": "foo"
+        },
+        {
+            "name": "event"
+        }
+    ],
+    "params": [
+        {
+            "name": "context1",
+            "schema":{
+                "type": "string"
+            }
+        },
+        {
+            "name": "context2",
+            "schema": {
+                "type": "number"
+            }
+        }
+    ],
+    "result": {
+        "name": "value",
+        "schema": {
+            "type": "boolean"
+        }
+    }
+}
+```
+
+Matching provider method:
+
+```json
+{
+    "name": "foo",
+    "tags": [
+        {
+            "name": "capabilities",
+            "x-provides": "xrn:firebolt:capabilities:example:foo"
+        },
+        {
+            "name": "event"
+        }
+    ],
+    "params": [
+        {
+            "name": "context1",
+            "schema":{
+                "type": "string"
+            }
+        },
+        {
+            "name": "context2",
+            "schema": {
+                "type": "number"
+            }
+        },
+        {
+            "name": "value",
+            "schema": {
+                "type": "boolean"
+            }
+        }
+    ]
+}
+```
+
+When a provider app calls a provider method mapped to an event the platform **MUST** ignore the notification if the app is not a [candidate app](#7-provider-candidates) for this capability.
```

Comment:
ignore or fail?

Reply:
I went with ignore. we might want to do stuff with the data in the future, or change the candidate policy. The app doesn't need to know we're not using them, it might encourage them to debug or stop calling, and it's not really their problem.